app/controllers/users_controller.rb

```diff
@@ -15,9 +15,65 @@ def create
     end
   end
 
+  def all_users
+    if admin?
+      @users = User.where.not(id: current_user.id)
+      render json: @users, each_serializer: UserSerializer
+    else
+      render json: { error: 'Not Allowed' }, status: :unauthorized
+    end
+  end
+
+  # Admin can update user info i.e (first_name, last_name, email, password and role)
+  def update_user_info
+    if admin?
+      check_email
+    else
+      render json: { error: 'Not Allowed' }, status: :unauthorized
+    end
+  end
```

Comment:
The controller looks great, but I have a question here. 

What is the purpose of this method? it calls the `checek_email` and `check_email` executes the `update_user`, can't we do it all at once in a single method?

Reply:
@aimalamiri, the check email is to check if the email a user is updating already exists. I didn't want to have a very big method because of rubocop line complexity

Reply:
@aimalamiri, I found a solution to make it  cleaner so that's fixed. You can check now